### ISSUE TITLE
False positive with multiline properties

### ISSUE THREAD

**Issue by EvgenyOrekhov** [NullVoxPopuli/eslint-plugin-decorator-position#313]
I stumbled upon this while working on a Stencil component.

Rule configuration: `"decorator-position/decorator-position": "error"` (default)

Code:

```js
export class MyComponent {
  @State() classes = ["toggle", this.checkedClass(), this.disabledClass()]
    .filter(Boolean)
    .join(" ");
}
```

Expected: no warnings (because by default property decorators should be inline)

Actual: `Expected @State to be on the line above`

**Comment by NullVoxPopuli** [comment]
is this still an issue? what config did you have?

**Comment by EvgenyOrekhov** [comment]
Yes, I still can reproduce it.

It seems it happens only when I add `"extends": ["plugin:prettier/recommended"]` to my config.

Here is the minimal reproducible case:

`.eslintrc.json`

```json
{
  "extends": ["plugin:prettier/recommended"],

  "parser": "@typescript-eslint/parser",

  "plugins": ["@typescript-eslint", "decorator-position"],

  "env": {
    "es2021": true
  },

  "parserOptions": {
    "sourceType": "module",
    "project": "./tsconfig.json"
  },

  "rules": {
    "decorator-position/decorator-position": "error"
  }
}

```

`tsconfig.json`

```json
{}
```

**Comment by NullVoxPopuli** [comment]
I made a reproduction here: https://github.com/NullVoxPopuli/eslint-plugin-decorator-position/pull/369
though, I had to tweak some things from the configs you gave

**Comment by NullVoxPopuli** [comment]
it looks like it's caused by the join being omitted from the line-length calculation

**Comment by NullVoxPopuli** [comment]
when I log the transformInfo and config in `placeDecoratorsAboveProperty` I see: 
```
  arity: 0,
  arityMatches: true,
  currentPositionMatchesIntent: false,
  needsTransform: true,
  inlineLength: 111,
  ifInlineWouldViolatePrettier: true,
  name: 'State',
  onDifferentLines: false,
  onSameLine: true,
  isMultiLineDecorator: false
} [ 'State', { intent: 'different-line' } ]
```

inlineLength is greater than default of 100 (or prettier default of 80), but the code is _not_ inline 🤔
